### PR TITLE
Add Composer Autoloading

### PR DESCRIPTION
**Pull request by zachluna**

### CODE REVIEW

File: tests/appengine/api/app_identity/AppIdentityServiceTest.php [hunk 1]
```diff
@@ -31,9 +31,11 @@ function apc_fetch($name, &$success) {
   $GLOBALS['fetch_calls']++;
   if (isset($GLOBALS['apc_fetch_result'])) {
     $item = array_shift($GLOBALS['apc_fetch_result']);
-    $result = $item['value'];
-    $success = $item['result'];
```
Comment (bshaffer) [inline tests/appengine/api/app_identity/AppIdentityServiceTest.php:35]: nit: You could do this instead:

```php
$result = $item['value'] ?? false;
$success = $item['result'] ?? false;
```

File: tests/appengine/util/ArrayUtilTest.php [hunk 2]
```diff
@@ -18,7 +18,7 @@
 
 use google\appengine\util\ArrayUtil;
 
-class ArrayUtilTest extends \PHPUnit_Framework_TestCase {
+class ArrayUtilTest extends \PHPUnit\Framework\TestCase {
```
Comment (bshaffer) [inline tests/appengine/util/ArrayUtilTest.php:21]: nit: import the `TesCase` class

```suggestion
use \PHPUnit\Framework\TestCase;

class ArrayUtilTest extends TestCase {
```

**Review by bshaffer — DISMISSED** [review]
Wow, looks great! Just two suggestions that are optional

File: composer.json [hunk 3]
```diff
@@ -18,6 +18,24 @@
         "php-coveralls/php-coveralls": "dev-master"
     },
     "autoload": {
-        "files": ["google/appengine/runtime/autoloader.php"]
+        "psr-4": {
+            "google\\": "src"
```
Comment (bshaffer) [inline composer.json:22]: One more thought! If we are removing the `datastore/` directory ( which I believe we are? ) we should also remove the `appengine` directory, move everything up into `src/` that is beneath it, and change this to:

```suggestion
            "google\\appengine\\": "src"
```

File: tests/appengine/api/app_identity/AppIdentityServiceTest.php (hunk 1 above)
Comment (zachluna) [reply tests/appengine/api/app_identity/AppIdentityServiceTest.php:35]: Ok I will update this, thanks!

File: composer.json (hunk 3 above)
Comment (zachluna) [reply composer.json:22]: Ok I will update this, thanks!

File: tests/appengine/util/ArrayUtilTest.php (hunk 2 above)
Comment (zachluna) [reply tests/appengine/util/ArrayUtilTest.php]: Updated this case for all unit tests, thanks!

**Review by bshaffer — APPROVED** [review]

**Pull request merged**